calcom/cal.com
# Fix credential creation flow for msteams app

## What does this PR do?

After some bug/issues on prod I took a look of what could be the error and found out some scenarios that wasn't handling well the credential update/creation for this app.

- Avoid duplicated credentials type office365_key
- Fix scope when renewing token

Fixes # (issue)

**Environment**: Staging(main branch) / Production

## Type of change

Loom: https://www.loom.com/share/cc6b2a5d42b44301bb2a37786e915b5b




## How should this be tested?

- Install MSteam app
- Add Office365video as Location for one of your event types
- Book with said event type many times with some time difference between them, so your current credential expires and gets renewed.
- You should not get error when creating an event after some time has passed.

## Checklist

- I haven't added tests that prove my fix is effective or that my feature works
